python/cudf/cudf/core/series.py

```diff
@@ -5216,3 +5218,47 @@ def isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
         result_col[equal_nulls] = True
 
     return Series(result_col, index=index)
+
+
+def factorize(values, sort=False, na_sentinel=-1):
+    """Encode the input values as integer labels
+
+    Parameters
+    ----------
+    values: Series, Index, or CuPy array
+        The data to be factorized.
+    na_sentinel : number
+        Value to indicate missing category.
```

Comment:
Should indicate default value in the docstring, also do we support `None` here similar to Pandas? If not we should throw if someone passes `None`.

Reply:
Added the default value. Apparently we do not support `na_sentinel=None` which is unfortunate. Raised https://github.com/rapidsai/cudf/issues/6946